### PR TITLE
[RFC] MSVC: Fix FindLuaJit libname for MSVC

### PR DESCRIPTION
**Pull request by sethjackson**
Cherry picked from #810. From equalsraf@94dcbe0 and equalsraf@bcd0579.

### CODE REVIEW

File: cmake/FindLuaJit.cmake [hunk 1]
```diff
@@ -32,8 +32,11 @@ if(LUAJIT_USE_STATIC)
 endif()
 
 list(APPEND LUAJIT_NAMES luajit-5.1)
+if(MSVC)
+  list(APPEND LUAJIT_NAMES lua51)
```
Comment (justinmk) [inline cmake/FindLuaJit.cmake]: Can anyone explain this change (and the similar one for libuv)?

Comment (sethjackson) [inline cmake/FindLuaJit.cmake]: cc @equalsraf. All I know is that without this MSVC doesn't find the library. As for why it's that why I have not investigated.

File: cmake/FindLuaJit.cmake [hunk 2]
```diff
@@ -32,8 +32,11 @@ if(LUAJIT_USE_STATIC)
 endif()
 
 list(APPEND LUAJIT_NAMES luajit-5.1)
+if(MSVC)
```
Comment (jszakmeister) [inline cmake/FindLuaJit.cmake]: Would more accurate check be to check `CMAKE_SYSTEM_NAME` for "Windows":

```cmake
if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
```

Comment (sethjackson) [inline cmake/FindLuaJit.cmake]: No. @equalsraf's branch compiles on MinGW too which doesn't need this one.

See equalsraf@bcd0579.

To be honest I should probably squash equalsraf@94dcbe0, and equalsraf@bcd0579
into this:

```
if(MSVC)
  list(APPEND LUAJIT_NAMES lua51)
elseif(MINGW)
  list(APPEND LUAJIT_NAMES libluajit)
else()
  list(APPEND LUAJIT_NAMES luajit-5.1)
endif()
```

Further for consistency with the rest of the build system if we want to check for Windows we do this: `if(WIN32)`.

@justinmk in response to your earlier question I'm not sure why the library names are different but you can see that `third-party/cmake/BuildLuajit.cmake` and `third-party/cmake/BuildLibuv.cmake` is also using those library names. The MinGW support for `luajit` is in equalsraf@b95df5e.

**Comment by justinmk** [comment]
Merged

**Pull request closed without merging**